### PR TITLE
First checkin for M365 Services Health SPFx webpart

### PR DESCRIPTION
**Pull request by HarminderSethi**
> By submitting this pull request, you agree to the [contribution guidelines](https://github.com/pnp/sp-dev-fx-webparts/blob/main/CONTRIBUTING.md)

> If you aren't familiar with how to contribute to open-source repositories using GitHub, or if you find the instructions on this page confusing, [sign up](https://forms.office.com/Pages/ResponsePage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUREZVRDVYUUJLT1VNRDM4SjhGMlpUNzBORy4u) for one of our [Sharing is Caring](https://pnp.github.io/sharing-is-caring/#pnp-sic-events) events. It's completely free, and we'll guide you through the process.

> To submit a pull request with multiple authors, make sure that at least one commit is a co-authored commit by adding a `Co-authored-by:` trailer to the commit's message. E.g.: `Co-authored-by: name <name@example.com>`

|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                               |
| New feature?    | no                             |
| New sample?     | yes                               |
| Related issues? | N/A |

## What's in this Pull Request?
1. Show the health status for all the M365 services
2. Complete details including all the updates for all the impacted services

### CODE REVIEW

File: samples/react-m365-services-health/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+## Applies to
+
+- [SharePoint Framework](https://learn.microsoft.com/sharepoint/dev/spfx/sharepoint-framework-overview)
+- [Microsoft 365 tenant](https://learn.microsoft.com/sharepoint/dev/spfx/set-up-your-development-environment)
+
+> Get your own free development tenant by subscribing to [Microsoft 365 developer program](http://aka.ms/m365devprogram)
+
+## Prerequisites
+
+> Any special pre-requisites?
```
Comment (hugoabernier) [inline samples/react-m365-services-health/README.md:41]: Please update the README.md to provide the required steps to allow permissions for the sample.

File: samples/react-m365-services-health/src/common/services/GraphService.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+	constructor(context: WebPartContext) {
+		this.context = context;
+	}
+
+	private getClient = async (): Promise<MSGraphClientV3> => {
+		return await this.context.msGraphClientFactory.getClient("3");
+	};
+
+	public getHealthOverviews = async (): Promise<any> => {
```
Comment (hugoabernier) [inline samples/react-m365-services-health/src/common/services/GraphService.ts:15]: Is there any way not to use `any` here?

File: samples/react-m365-services-health/src/common/services/HelperService.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
+export namespace HelperService {
```
Comment (hugoabernier) [inline samples/react-m365-services-health/src/common/services/HelperService.ts:1]: Consider https://typescript-eslint.io/rules/no-namespace/

File: samples/react-m365-services-health/src/common/services/HelperService.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	export const getProductIcon = (value: string): string => {
+		let icon: string = "Product";
+		switch (value?.toLowerCase()) {
+			case "microsoft intune":
+			case "mobile device management for office 365":
+			case "identity service":
+			case "microsoft 365 apps":
+			case "microsoft 365 defender":
+			case "microsoft 365 for the web":
+			case "microsoft mefender for cloud apps":
```
Comment (hugoabernier) [inline samples/react-m365-services-health/src/common/services/HelperService.ts:26]: Should this be `defender`?

File: samples/react-m365-services-health/src/webparts/m365ServicesHealth/services/ListViewHelperService.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import { IServiceHealthOverviewItem } from "./../interfaces/ServiceHealthModels";
+import { IColumn } from "@fluentui/react";
+import { ServiceHealth, ServiceHealthIssue } from "@microsoft/microsoft-graph-types";
+
+export namespace ListViewHelperService {
```
Comment (hugoabernier) [inline samples/react-m365-services-health/src/webparts/m365ServicesHealth/services/ListViewHelperService.ts:5]: Consider https://typescript-eslint.io/rules/no-namespace/

File: samples/react-m365-services-health/README.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
+# Microsoft 365 Services Health
```
Comment (hugoabernier) [inline samples/react-m365-services-health/README.md:1]: Try renaming it "Service Health for Microsoft 365" to comply with Microsoft branding guidelines.

**Review by hugoabernier — CHANGES_REQUESTED** [review]
@HarminderSethi this sample looks great, please make the changes listed in this review so we can merge the PR.

**Review by hugoabernier — APPROVED** [review]

**Comment by HarminderSethi** [comment]
Thanks @hugoabernier 😊

**Comment by hugoabernier** [comment]
Thank you @HarminderSethi for this sample. Would you be interested in demoing on the April 20th SPFx community call? You'd have 10-13 minutes to demo your web part and review code highlights.

**Comment by HarminderSethi** [comment]
@hugoabernier Yes, I would be interested in demoing in SPFx community call. Thank you for the opportunity to showcase my web part.

**Comment by hugoabernier** [comment]
Perfect! I have you confirmed for April 20th. Can't wait to see your demo!

**Pull request merged**